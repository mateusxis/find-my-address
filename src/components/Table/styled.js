import styled from 'styled-components';

export const STable = styled.table`
  width: 100%;
`;

export const STitle = styled.h2`
  font-size: 26px;
  margin: 20px 0;
  text-align: center;
  color: #ffffff;
`;
