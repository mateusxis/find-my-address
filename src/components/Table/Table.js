import { STable, STitle } from './styled';
import ColumnTable from './ColumnTable';
import HeaderTable from './HeaderTable';
import RowTable from './RowTable';
import hashifyKey from '../../utils/hashifyKey';

const Table = ({ data = [], title = '' }) => {
  return data.length > 0 ? (
    <>
      {title && <STitle>{title}</STitle>}
      <STable>
        <HeaderTable>
          <ColumnTable label="Logradouro" className="c1">
            Logradouro
          </ColumnTable>
          <ColumnTable label="Bairro" className="c2">
            Bairro
          </ColumnTable>
          <ColumnTable label="Cidade" className="c3">
            Cidade
          </ColumnTable>
          <ColumnTable label="Estado" className="c4">
            Estado
          </ColumnTable>
          <ColumnTable label="CEP" className="c5">
            CEP
          </ColumnTable>
        </HeaderTable>
        {data.map((value, index) => (
          <RowTable key={hashifyKey(`${value.zipCode}_${index}`)}>
            <ColumnTable label="Logradouro" className="c1">
              {value.address}
            </ColumnTable>
            <ColumnTable label="Bairro" className="c2">
              {value.district}
            </ColumnTable>
            <ColumnTable label="Cidade" className="c3">
              {value.city}
            </ColumnTable>
            <ColumnTable label="Estado" className="c4">
              {value.state}
            </ColumnTable>
            <ColumnTable label="CEP" className="c5">
              {value.zipCode}
            </ColumnTable>
          </RowTable>
        ))}
      </STable>
    </>
  ) : null;
};

export default Table;
