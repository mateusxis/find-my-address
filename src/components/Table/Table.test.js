import renderer from 'react-test-renderer';

import Table from './Table';

describe('Table', () => {
  it('should render Table', () => {
    const tree = renderer
      .create(
        <Table
          data={[
            {
              address: 'Street One',
              city: 'New York',
              zipCode: '12345-000',
              district: 'Queen',
              state: 'New York'
            }
          ]}
          title="My table"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
