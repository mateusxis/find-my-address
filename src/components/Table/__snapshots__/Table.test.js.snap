// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render Table 1`] = `
Array [
  <h2
    className="sc-gsDKAQ gwMJsV"
  >
    My table
  </h2>,
  <table
    className="sc-bdvvtL eukWAO"
  >
    <tr
      className="sc-hKwDye bFMAfG"
    >
      <td
        className="sc-dkPtRN ePeDwK c1"
        data-label="Logradouro"
      >
        Logradouro
      </td>
      <td
        className="sc-dkPtRN ePeDwK c2"
        data-label="Bairro"
      >
        Bairro
      </td>
      <td
        className="sc-dkPtRN ePeDwK c3"
        data-label="Cidade"
      >
        Cidade
      </td>
      <td
        className="sc-dkPtRN ePeDwK c4"
        data-label="Estado"
      >
        Estado
      </td>
      <td
        className="sc-dkPtRN ePeDwK c5"
        data-label="CEP"
      >
        CEP
      </td>
    </tr>
    <tr
      className="sc-eCImPb Brvou"
    >
      <td
        className="sc-dkPtRN ePeDwK c1"
        data-label="Logradouro"
      >
        Street One
      </td>
      <td
        className="sc-dkPtRN ePeDwK c2"
        data-label="Bairro"
      >
        Queen
      </td>
      <td
        className="sc-dkPtRN ePeDwK c3"
        data-label="Cidade"
      >
        New York
      </td>
      <td
        className="sc-dkPtRN ePeDwK c4"
        data-label="Estado"
      >
        New York
      </td>
      <td
        className="sc-dkPtRN ePeDwK c5"
        data-label="CEP"
      >
        12345-000
      </td>
    </tr>
  </table>,
]
`;
